File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        vector<int>ans;
        ans.push_back(0);
        if(gain.size()==0){
            return 0;
        }
        ans.push_back(gain[0]);

        for(int i=1;i<gain.size();i++){
          int num=gain[i]+ans[i];
          ans.push_back(num);
        }

         
        int max=INT_MIN;
        for(int i=0;i<ans.size();i++){
           if(max<ans[i]){
            max=ans[i];
           }
        }
        return max;
    }
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        List<Integer>ans=new ArrayList<>();
        ans.add(0);
        ans.add(gain[0]);
         if (gain.length == 0) return 0;

        for(int i=1;i<gain.length;i++){
         int num=gain[i]+ans.get(i);
         ans.add(num);
        }
        int max=Integer.MIN_VALUE;
        for(int i=0;i<ans.size();i++){
           if(max<ans.get(i)){
            max=ans.get(i);
           }
        }
        return max;
    }
}